ccl/sqlproxyccl/acl: TestParsingErrorHandling failed

ccl/sqlproxyccl/acl.TestParsingErrorHandling [failed](https://mesolite.cluster.engflow.com/invocations/default/4ea9d28e-ba7f-4b1d-bcfd-0c372a0ecf66?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NxbHByb3h5Y2NsL2FjbDphY2xfdGVzdA==) on release-25.2.11-rc @ [0d9e4ae4e7e3c35adea65f020bfec48628482761](https://github.com/cockroachdb/cockroach/commits/0d9e4ae4e7e3c35adea65f020bfec48628482761):


```
=== RUN   TestParsingErrorHandling
E260113 06:31:33.458278 54 ccl/sqlproxyccl/acl/file.go:106  [-] 14  WatchList daemon stopped: context canceled
--- FAIL: TestParsingErrorHandling (0.44s)
=== RUN   TestParsingErrorHandling/error_after_update
    file_test.go:561: should not have gotten a new controller
--- FAIL: TestParsingErrorHandling/error_after_update (0.19s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146702 ccl/sqlproxyccl/acl: TestParsingErrorHandling failed [C-test-failure O-robot T-serverless branch-release-25.2]
- #143472 ccl/sqlproxyccl/acl: TestParsingErrorHandling failed [C-test-failure O-robot T-serverless branch-release-24.3]
</p>
</details>
/cc @cockroachdb/sql-foundations @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParsingErrorHandling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58636

--- a/pkg/ccl/sqlproxyccl/acl/file_test.go
+++ b/pkg/ccl/sqlproxyccl/acl/file_test.go
@@ -558,9 +558,7 @@ func TestParsingErrorHandling(t *testing.T) {
 		testutils.SucceedsSoon(t, func() error {
 			select {
 			case <-next:
-				t.Fatal("should not have gotten a new controller")
-				// We need to return something to make the compiler happy, but t.Fatal will end execution.
-				return nil
+				return fmt.Errorf("Received new controller; discarding")
 			default:
 				errorCount := errorCountMetric.Value()
 				// If error count isn't one, then it hasn't happened yet.

